sycl/source/detail/device_impl.cpp

```diff
@@ -546,6 +546,57 @@ bool device_impl::has(aspect Aspect) const {
             sizeof(pi_bool), &support, nullptr) == PI_SUCCESS;
     return call_successful && support;
   }
+  case aspect::ext_oneapi_bindless_sampled_image_fetch_1d_usm: {
+    pi_bool support = PI_FALSE;
+    bool call_successful =
+        getPlugin()->call_nocheck<detail::PiApiKind::piDeviceGetInfo>(
```

Comment:
It would be best to switch to using `call` over `call_nocheck` as pointed out in [this thread](https://github.com/intel/llvm/pull/12996#discussion_r1547622631). Please do the same for all these queries.

Reply:
I've updated all queries in this PR to use `call`

Reply:
Reviving this. Converting to using `call` causes CI failures as soon as the E2E test stage starts on L0, HIP, and OpenCL (GPU/CPU).

I suspect that since these backends don't support all of our queries, they throw an error, and that error is not caught as soon as the E2E tests begin.